nistats/first_level_model.py

```diff
@@ -0,0 +1,561 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""
+This module presents an interface to use the glm implemented in
+nistats.regression.
+
+It contains the GLM and contrast classes that are meant to be the main objects
+of fMRI data analyses.
+"""
+
+from warnings import warn
+import time
+import sys
+import os
+
+import numpy as np
+import glob
+from nibabel import Nifti1Image
+
+from sklearn.base import BaseEstimator, TransformerMixin, clone
+from sklearn.externals.joblib import Memory
+from nilearn._utils.niimg_conversions import check_niimg
+from nilearn._utils import CacheMixin
+from nilearn._utils.class_inspect import get_params
+from nilearn.input_data import NiftiMasker
+from sklearn.externals.joblib import Parallel, delayed
+
+from .regression import OLSModel, ARModel, SimpleRegressionResults
+from .design_matrix import make_design_matrix
+from .contrasts import _summary_contrast
+from .utils import _basestring, _check_run_tables
+
+
+def mean_scaling(Y, axis=0):
+    """Scaling of the data to have percent of baseline change along the
+    specified axis
+
+    Parameters
+    ----------
+    Y : array of shape (n_time_points, n_voxels)
+       The input data.
+
+    Returns
+    -------
+    Y : array of shape (n_time_points, n_voxels),
+       The data after mean-scaling, de-meaning and multiplication by 100.
+
+    mean : array of shape (n_voxels,)
+        The data mean.
+    """
+    mean = Y.mean(axis=axis)
+    if (mean == 0).any():
+        warn('Mean values of 0 observed.'
+             'The data have probably been centered.'
+             'Scaling might not work as expected')
+    mean = np.maximum(mean, 1)
+    Y = 100 * (Y / mean - 1)
+    return Y, mean
+
+
+def _ar_model_fit(X, val, Y):
```

Comment:
I think we should avoid such one-liners. It makes debugging and reviewing harder and does not really help.